### ISSUE TITLE
Missing TS type definition for relay config

### ISSUE THREAD

**Issue by MarceloPrado** [facebook/relay#4348]
Jest offer users typed definitions for their config files:
```js
/** @type {import('jest').Config} */
const config = {
  verbose: true,
};

module.exports = config;
```

As explained in their [docs](https://jestjs.io/docs/configuration), you can write your docs in JS and use JSDocs to get type inference. Another approach is to write the config file in TS and directly import the `Config` type.

Would the Relay team consider offering something similar? It would improve the DX when configuring the compiler. The current approach requires me to open the `relay-compiler` README to manually ensure I'm using the right options.

**Comment by maraisr** [comment]
We did have a contribution to [our types](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64448) to add this, but was abandoned.

I'll raise a PR to fix this. Sorry about that, that PR snuck through and didnt notice.

Wondering though, would you @captbaritone be open to have this config file included in the `relay-compiler` package, or keep things over at DT for now?

**Comment by captbaritone** [comment]
I love the idea, and this is something I would love for us to have. That said, I think there is at least one questions we should answer first:

How can we ensure the types file gets updated every time the compiler itself gets updated?

**Comment by maraisr** [comment]
Yeah I think there is a bigger story here with typescript in general, but at least for today — perhaps we can use the existing config types to codegen the typescript types? If you have ideas, happy yo PoC it.

**Comment by EloB** [comment]
To his two flies in one hit than also upload a JSON schema to https://www.schemastore.org/json/ to make VSCode and a bunch of editors autocomplete it. Maybe promote a name for that json config like `relay.config.json` or something.

**Comment by alecmev** [comment]
It's not released yet, but there's a schema now, added in https://github.com/facebook/relay/pull/4723:

```jsonc
{
  "$schema": "https://github.com/facebook/relay/raw/9f314a080274960ccd9257b66c47b3c6f3d74956/compiler/relay-compiler-config-schema.json",
  // ...
}
```

**Comment by MarceloPrado** [comment]
Fixed by #4724

**Comment by tobias-tengler** [comment]
The original issue isn't really solved though, right? You now have support for JSON files, but if you're using the JS config loader you still don't have any TypeScript types (unless you generate them yourself based on the JSON schema)

**Comment by MarceloPrado** [comment]
@tobias-tengler you're right, sorry for closing this. It still makes sense to provide something at the Relay land for JS configs.
